action.yml

```diff
@@ -1,60 +1,188 @@
     required: false
 
 runs:
-  using: 'composite'
+  using: "composite"
   steps:
-  - name: Create a Virtual Environment
-    run: |
-      pip install virtualenv
-      virtualenv venv
-    shell: bash
-
-  - name: Install Tutor
-    run: |
-      source venv/bin/activate
-      if [ "$INPUT_TUTOR_VERSION" = "nightly" ]; then
-        git clone --branch=nightly --depth=1 https://github.com/overhangio/tutor.git
-        chmod 777 tutor -R
-        pip install -e "./tutor[full]"
-      else
-        pip install "tutor$INPUT_TUTOR_VERSION"
-      fi
-    shell: bash
-    env:
-      INPUT_TUTOR_VERSION: ${{ inputs.tutor_version }}
-
-  - name: Configure Tutor and Launch
-    run: |
-      source venv/bin/activate
-      TUTOR_ROOT="$(pwd)" tutor --version
-      TUTOR_ROOT="$(pwd)" tutor config save
-      TUTOR_ROOT="$(pwd)" tutor mounts add lms,cms,lms-worker,cms-worker:$(pwd)/$INPUT_APP:/openedx/$INPUT_APP
-      chmod 777 . -R
-      if [ "$INPUT_TUTOR_VERSION" = "nightly" ]; then
-        TUTOR_ROOT="$(pwd)" tutor images build openedx
-      fi
-      TUTOR_ROOT="$(pwd)" tutor local launch -I
-    shell: bash
-    env:
-      INPUT_APP: ${{ inputs.app_name }}
-      INPUT_TUTOR_VERSION: ${{ inputs.tutor_version }}
-
-  - name: Run integration tests in lms
-    if: ${{ inputs.shell_file_to_run }}
-    run: |
-      source venv/bin/activate
-      TUTOR_ROOT="$(pwd)" tutor local run lms bash $INPUT_SHELL_FILE
-    shell: bash
-    env:
-      INPUT_SHELL_FILE: /openedx/${{ inputs.app_name }}/${{ inputs.shell_file_to_run }}
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        path: ${{ inputs.app_name }}
+
+    - name: Adjust permissions to execute Tutor commands
+      run: |
+        chmod 777 . -R
+      shell: bash
+
+    - name: Set Tutor environment variables
+      run: |
+        echo "LMS_HOST=local.edly.io" >> "$GITHUB_ENV"
+        echo "CMS_HOST=studio.local.edly.io" >> "$GITHUB_ENV"
+        echo "TUTOR_ROOT=$(pwd)" >> "$GITHUB_ENV"
+        echo "TUTOR_PLUGINS_ROOT=$(pwd)/plugins/" >> "$GITHUB_ENV"
+      shell: bash
+
+    - name: Create virtualenvs
+      run: |
+        echo "Creating isolated venv for Tutor"
+        python -m venv .tutor_venv
+        echo "Creating isolated venv to run the integration tests"
+        python -m venv .tests_venv
+      shell: bash
+
+    - name: Install and prepare Tutor
+      run: |
+        source .tutor_venv/bin/activate
+
+        if [ "$INPUT_TUTOR_VERSION" = "nightly" ]; then
+          git clone --branch=nightly --depth=1 https://github.com/overhangio/tutor.git
+          pip install -e ./tutor
+        else
+          pip install "tutor$INPUT_TUTOR_VERSION"
+        fi
+
+        tutor config save --set LMS_HOST=$LMS_HOST --set CMS_HOST=$CMS_HOST
+        tutor local launch -I
+      shell: bash
+      env:
+        INPUT_TUTOR_VERSION: ${{ inputs.tutor_version }}
```

Comment:
I think moving this up after `name` would make the step easier to read.

Reply:
Done!